crowbar_framework/lib/crowbar/upgrade.rb

```diff
@@ -0,0 +1,70 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+module Crowbar
+  class Upgrade
+    attr_accessor :data
+
+    def initialize(backup)
+      @backup = backup
+      @data = @backup.data
+      @version = @backup.version
+    end
+
+    def upgrade
```

Comment:
I find the name a bit confusing: this is upgrading the data in a backup to something we understand, so should we name this `upgrade_backup`?